block_windows.go

```diff
@@ -0,0 +1,218 @@
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package ghw
+
+import (
+	"strings"
+
+	"github.com/StackExchange/wmi"
+)
+
+type Win32_DiskDrive struct {
+	Caption           string
+	CreationClassName string
+	Description       string
+	DeviceID          string
+	Index             uint32 // Used to link with partition
+	InterfaceType     string
+	Manufacturer      string
+	MediaType         string
+	Model             string
+	Name              string
+	Partitions        int32
+	SerialNumber      string
+	Size              uint64
+	TotalCylinders    int64
+	TotalHeads        int32
+	TotalSectors      int64
+	TotalTracks       int64
+	TracksPerCylinder int32
+}
+
+/*type Win32_DiskDriveToDiskPartition struct {
+}*/
+
+type Win32_DiskPartition struct {
+	Access            uint16
+	BlockSize         uint64
+	Caption           string
+	CreationClassName string
+	Description       string
+	DeviceID          string
+	DiskIndex         uint32 // Used to link with Disk Drive
+	Index             uint32
+	Name              string
+	Size              int64
+	SystemName        string
+	Type              string
+}
+
+type Win32_LogicalDiskToPartition struct {
+	Antecedent string
+	Dependent  string
+}
+
+type Win32_LogicalDisk struct {
+	Caption           string
+	CreationClassName string
+	Description       string
+	DeviceID          string
+	FileSystem        string
+	FreeSpace         uint64
+	Name              string
+	Size              uint64
+	SystemName        string
+}
+
+func (ctx *context) blockFillInfo(info *BlockInfo) error {
+	// Getting disk drives from WMI
+	var win32DiskDriveDescriptions []Win32_DiskDrive
+	q1 := wmi.CreateQuery(&win32DiskDriveDescriptions, "")
+	if err := wmi.Query(q1, &win32DiskDriveDescriptions); err != nil {
+		return err
+	}
+
+	// Getting disk partitions from WMI
+	var win32DiskPartitionDescriptions []Win32_DiskPartition
+	q2 := wmi.CreateQuery(&win32DiskPartitionDescriptions, "")
+	if err := wmi.Query(q2, &win32DiskPartitionDescriptions); err != nil {
+		return err
+	}
+
+	// Getting links between logical disks and partitions from WMI
+	var win32LogicalDiskToPartitionDescriptions []Win32_LogicalDiskToPartition
+	q3 := wmi.CreateQuery(&win32LogicalDiskToPartitionDescriptions, "")
+	if err := wmi.Query(q3, &win32LogicalDiskToPartitionDescriptions); err != nil {
+		return err
+	}
+
+	// Getting logical disks from WMI
+	var win32LogicalDiskDescriptions []Win32_LogicalDisk
+	q4 := wmi.CreateQuery(&win32LogicalDiskDescriptions, "")
+	if err := wmi.Query(q4, &win32LogicalDiskDescriptions); err != nil {
+		return err
+	}
```

Comment:
for each of the above, please create a separate helper function that returns the slice of appropriate structs, where the helper function uses a const string `cmi.Query()` instead of the `cmi.CreateQuery()` function.

Example for the disk drives:

```go
const (
        wqlDiskDrives = `
SELECT Caption, CreationClassName, Description, DeviceID,
    Index, InterfaceType. Manufacturer, MediaType, Model, Name,
    SerialNumber, Size, TotalCylinders, TotalHeads, TotalSectors,
    TotalTracks, TracksPerCylinder
FROM Win32_DiskDrive`
)

func (ctx *context) blockFillInfo(info *BlockInfo) error {
        win32DiskDriveDescriptions, err := getDiskDrives()
        if err != nil {
                return err
        }
...
}

func getDiskDrives() ([]win32DiskDrive, error) {
	// Getting disk drives from WMI
	var win32DiskDriveDescriptions []win32DiskDrive
	if err := wmi.Query(wqlDiskDrives, &win32DiskDriveDescriptions); err != nil {
		return nil, err
	}
        return win3232DiskDriveDescriptions, nil
}
```